extensions/basic_auth/config.proto

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+// $title: Basic Auth Config
+// $description: Configuration for Basic Auth Filter.
+// $weight: 20
+
+// The following is an example of a configuration that restricts access to
+// given path prefixes, request methods and credentials.
+// {
+//   "basic_auth_rules": [
+//     {
+//       "request_path":"/api/reviews/pay",
+//       "match": "exact",
+//       "request_methods":[ "GET", "POST" ],
+//       "credentials":[ "admin:admin", "admin2:admin2" ]
+//     },
+//     {
+//       "request_path":"/api",
```

Comment:
The example does not seem to match the schema.

Reply:
I added the submessage "match" to be able to know which match pattern to apply. What part doesn't match the schema?

Reply:
oh i see, you mean that instead of being `"request_path": "/api"` it should be `"prefix": "/api"` , right?